### ISSUE TITLE
Display of negative int8 tensor values

### ISSUE THREAD

**Issue by mdouze** [pytorch/pytorch#52146]
## 🐛 Bug

Display of int8 matrix seems broken. 

## To Reproduce

```
>>> import torch
>>> a = torch.tensor([1, -10], dtype=torch.int8)
>>> a[1]
tensor(-10, dtype=torch.int8)
>>> a
tensor([  1, 246], dtype=torch.int8)
```

## Expected behavior

I expect the tensor to be written as ` tensor([  1, -10], dtype=torch.int8)`

This works fine on pytorch 1.7.1 on x64.

## Environment
```
(pytorch_venv) matthijs@ip-10-100-175-26:~/efficient-deit/deit_lite$ python collect_env.py
Collecting environment information...
PyTorch version: 1.8.0.dev20201210
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (aarch64)
GCC version: (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2

Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.0
[pip3] torch==1.8.0.dev20201210
[pip3] torchvision==0.2.2.post3
[conda] Could not collect
```

**Comment by albanD** [comment]
Hi,

I couldn't reproduce with latest nightly on a colab machine:
```
1.9.0.dev20210211+cpu
tensor([  1, -10], dtype=torch.int8)
```

Could you check if you're still seeing this with latest nightly please?

**Comment by mdouze** [comment]
Thanks for checking! 

I am running this on aarch64. This is with the latest lightly build for that architecture (from Dec 10, 2020 apparently). Installed from: 

https://download.pytorch.org/whl/nightly/cpu/torch-1.8.0.dev20201210-cp38-cp38-linux_aarch64.whl

**Comment by albanD** [comment]
triage review to see if someone else should be cc-ed here.

**Comment by malfet** [comment]
Likely a minor compiler issue (char is signed on x86, but unsigned on arm)
Although I can't repro on M1, will check on aarch64 later:
```
% python3 -c "import torch;print(torch.__version__);a=torch.tensor([1, -10], dtype=torch.int8);print(a)"
1.8.0a0+109bc10
tensor([  1, -10], dtype=torch.int8)
```

P.S. Added myself to watch module: arm label

**Comment by malfet** [comment]
Looks like problem is in `tensor_to_list` method:
```
 $ python3 -c "import torch;a=torch.tensor([-1], dtype=torch.int8);print(a.tolist())"
[255]

```

**Comment by malfet** [comment]
https://download.pytorch.org/whl/nightly/cpu/torch-1.9.0.dev20210224-cp38-cp38-linux_aarch64.whl should no longer have this problem
